### PR TITLE
Misc DB changes following review

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        uniqueColumns: resource_uid
+    - addPrimaryKey:
+        tableName: artifact_last_checked
+        columnNames: artifact_uid
+    - addPrimaryKey:
+        tableName: delivery_config_last_checked
+        columnNames: delivery_config_uid
+    - addPrimaryKey:
+        tableName: resource_last_checked
+        columnNames: resource_uid
```
Comment (robfletcher) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:26]: Replace unique keys with primary keys

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - renameColumn:
+        tableName: event
+        oldColumnName: uid
+        newColumnName: ref
+        columnDataType: varchar(255)
```
Comment (robfletcher) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:31]: I think the intent of this column was causing confusion (it confused me). Renaming & will add a separate `uid` for the event itself in a following PR.

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - dropIndex:
+        tableName: artifact_versions
+        indexName: artifact_versions_artifact_idx
+    - dropIndex:
+        tableName: delivery_config_artifact
+        indexName: delivery_config_artifact_unique_idx
+    - dropIndex:
+        tableName: environment_resource
+        indexName: environment_resource_unique_idx
```
Comment (robfletcher) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:40]: Dropped redundant indices

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    - sql:
+        comment: make JSON columns last in the table for performance
+        sql: |
+          alter table event modify column json json after scope;
+          alter table delivery_artifact modify column details json after reference;
```
Comment (robfletcher) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:45]: Moved columns to end of table

**Comment by emjburns** [comment]
Do you have any worry about rolling these all out at once?

**Comment by robfletcher** [comment]
> Do you have any worry about rolling these all out at once?

I deliberately didn't handle the couple of wider-reaching changes in this PR. I think these ones are pretty minor.

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - addPrimaryKey:
+        tableName: artifact_last_checked
+        columnNames: artifact_uid
+    - addPrimaryKey:
+        tableName: delivery_config_last_checked
+        columnNames: delivery_config_uid
+    - addPrimaryKey:
+        tableName: resource_last_checked
+        columnNames: resource_uid
+    - renameColumn:
```
Comment (luispollo) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:27]: This should be a separate changeset in this changelog, no?

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml (hunk 3 above)
Comment (luispollo) [reply keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:40]: These should be a separate changeset in this changelog, no?

Comment (luispollo) [reply keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:40]: P.S. I believe there was a redundant index in `event` as well (`event_scope_uid_idx`).

File: keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        uniqueColumns: artifact_uid
+    - dropUniqueConstraint:
+        tableName: delivery_config_last_checked
+        constraintName: delivery_config_uid
+        uniqueColumns: delivery_config_uid
+    - dropUniqueConstraint:
+        tableName: resource_last_checked
+        constraintName: resource_uid
+        uniqueColumns: resource_uid
+    - addPrimaryKey:
```
Comment (marchello2000) [inline keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:18]: i assume this was generated by jooq? Will it just convert these multiple primary keys to a single composite primary key?

Comment (marchello2000) [reply keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:18]: I am not sure, but I think the syntax here is wrong:
https://www.liquibase.org/documentation/changes/add_primary_key.html

i think it's supposed to be something like this:
```yaml
    - addPrimaryKey:
        tableName: delivery_config_last_checked
        columnNames: artifact_last_checked, delivery_config_uid, resource_last_checked
```
(instead of 3 adds)
also, assuming the order these columns is what you want - since they will comprise the PK index and will serve as the main lookup.

Comment (marchello2000) [reply keel-sql/src/main/resources/db/changelog/20200511-misc-platform-review-changes.yml:18]: Sorry, ignore me, i am an idiot... these are different tables... 🤦

**Comment by robfletcher** [comment]
I separated out the changesets

**Comment by robfletcher** [comment]
> this makes sense to me, but going to caveat it with "this is the second time I am looking at keel source so not sure if anything else needs to change", but I assume since the model is encoded in code and if it compiles - it runs :)

Yeah, I did the simpler changes all together in this PR. There are a couple of more involved ones you guys suggested that I'll tackle separately.
